From b6266f400f221b0a53ac05a131b24d0e0a19cb22 Mon Sep 17 00:00:00 2001
From: Ayyappa Ch <ayyappa.chandolu@amd.com>
Date: Fri, 27 Jan 2017 12:39:11 +0530
Subject: [PATCH 1607/1722] drm/amdgpu: Get user pages in non-current task

9232dd74ec9939a4ad622a73117668a02ffdec7f commit shows below description:
get_user_pages is changed to be called from current task where mm is being
operated on. get_user_pages_remote is introduced when the target mm is not
the current task's. Adjust amdgpu_ttm_tt_get_user_pages so it'll get user
pages in both cases.

4.4 kernel do not support get_user_pages_remote. Existing get_user_pages
will take care of current and other  task. Modified code to support
all tasks by bassing usertask from gtt struct.

Signed-off-by: Ayyappa Ch <ayyappa.chandolu@amd.com>
---
 drivers/gpu/drm/amd/amdgpu/amdgpu_ttm.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/amd/amdgpu/amdgpu_ttm.c b/drivers/gpu/drm/amd/amdgpu/amdgpu_ttm.c
index c44fed3..94da1dc 100644
--- a/drivers/gpu/drm/amd/amdgpu/amdgpu_ttm.c
+++ b/drivers/gpu/drm/amd/amdgpu/amdgpu_ttm.c
@@ -647,7 +647,7 @@ int amdgpu_ttm_tt_get_user_pages(struct ttm_tt *ttm, struct page **pages)
 		list_add(&guptask.list, &gtt->guptasks);
 		spin_unlock(&gtt->guptasklock);
 
-		r = get_user_pages(current, current->mm, userptr, num_pages,
+		r = get_user_pages(gtt->usertask, gtt->usertask->mm, userptr, num_pages,
 				   write, 0, p, NULL);
 
 		spin_lock(&gtt->guptasklock);
-- 
2.7.4

